pandas/tests/frame/test_combine_concat.py

```diff
@@ -763,3 +763,14 @@ def test_concat_datetime_datetime64_frame(self):
 
         # it works!
         pd.concat([df1, df2_obj])
+
+
+class TestDataFrameUpdate(TestData):
+
+    def test_update_nan(self):
+        # #15593 #15617
+        expected = df1 = DataFrame({'A': [1.0, 2, 3], 'B': date_range('2000', periods=3)})
+        df2 = DataFrame({'A': [None, 2, 3]})
+        df1.update(df2, overwrite=False)
+        # it works!
+        tm.assert_frame_equal(df1, expected)
```

Comment:
Can you add a second test with 

```python
df1 = DataFrame({'A': [1,None,3], 'B': date_range('2000', periods=3)})
df1.update(df2, overwrite=False)
```

just to make sure we cover the original issue? You can do it in this same test, and use the same `expected`